scripts/install.in

```diff
@@ -25,18 +25,44 @@ require_util() {
 }
 
 case "$(uname -s).$(uname -m)" in
-    Linux.x86_64) system=x86_64-linux; hash=@binaryTarball_x86_64-linux@;;
-    Linux.i?86) system=i686-linux; hash=@binaryTarball_i686-linux@;;
-    Linux.aarch64) system=aarch64-linux; hash=@binaryTarball_aarch64-linux@;;
-    Darwin.x86_64) system=x86_64-darwin; hash=@binaryTarball_x86_64-darwin@;;
-    # eventually maybe: system=arm64-darwin; hash=@binaryTarball_arm64-darwin@;;
-    Darwin.arm64) system=x86_64-darwin; hash=@binaryTarball_x86_64-darwin@;;
+    Linux.x86_64)
+        hash=@tarballHash_x86_64-linux@
+        store_path=@tarballPath_x86_64-linux@
+        system=x86_64-linux
+        ;;
+    Linux.i?86)
+        hash=@tarballHash_i686-linux@
+        store_path=@tarballPath_i686-linux@
+        system=i686-linux
+        ;;
+    Linux.aarch64)
+        hash=@tarballHash_aarch64-linux@
+        store_path=@tarballPath_aarch64-linux@
+        system=aarch64-linux
+        ;;
+    Darwin.x86_64)
+        hash=@tarballHash_x86_64-darwin@
+        store_path=@tarballPath_x86_64-darwin@
+        system=x86_64-darwin
+        ;;
+    Darwin.arm64)
+        # eventually maybe: system=arm64-darwin
+        system=x86_64-darwin
+        store_path=@tarballPath_x86_64-darwin@
+        hash=@binaryTarball_x86_64-darwin@
+        ;;
     *) oops "sorry, there is no binary distribution of Nix for your platform";;
 esac
 
-url="https://releases.nixos.org/nix/nix-@nixVersion@/nix-@nixVersion@-$system.tar.xz"
+# Use this command-line option to fetch the tarballs using nar-serve.
+if "${1:---nar-serve}"; then
+    url=${2}${store_path}
```

Comment:
Could we settle on URL pattern that is a bit more descriptive?

Cachix supports `/serve/{storehash}/{filepath}`

Reply:
Is the `{storehash}` just the hash, with the derivation name stripped?

Reply:
nar-serve supports `/nix/store/{storehash}/{filepath}` as well

Reply:
it's just the hash.

I think prefixing with `/nix/store` is a bad idea for two reasons:

- store prefix is configurable, so that pattern can't be used in Cachix
- it doesn't have a prefix for the action, in case the backend supports more than just serving nars

Reply:
Technically, you could also add the support for `/serve/${storePath}` to Cachix. Cachix knows about its store prefix, and the script can take a path as part of the base URL.

Reply:
fixed by 7c083f1